File: jpa-sample/src/main/java/com/example/demo/UserController.java
package com.example.demo;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
public class UserController {

    @Autowired
    private UserService userService;

    @GetMapping("/users")
    public List<User> findAllUsers() {
        return userService.findAllUsers();
    }

    @PostMapping("/user")
    public void saveUser() {
        User user = new User();
        user.setName("Bob");
        user.setEmail("hoge@example.com");
        userService.saveUser(user);
    }
}

File: multi-db/src/main/kotlin/com/example/multidb/controller/UserController.kt
package com.example.multidb.controller

import com.example.multidb.domain.user.User
import com.example.multidb.service.UserService
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class UserController(
    private val userService: UserService
) {

    @GetMapping("/users")
    fun getAllUser(): List<User> {
        return userService.getAllUser()
    }

    @PostMapping("/users")
    fun saveUser(
        @RequestBody request: UserRequestBody
    ): User {
        return userService.saveUser(request.name, request.age)
    }


    data class UserRequestBody(
        val name: String,
        val age: Int
    )
}